cockatrice/src/dlg_filter_games.cpp

```diff
@@ -230,9 +237,13 @@ int DlgFilterGames::getMaxPlayersFilterMax() const
     return maxPlayersFilterMaxSpinBox->value();
 }
 
-int DlgFilterGames::getMaxGameAgeComboBoxIndex() const
+const QTime &DlgFilterGames::getMaxGameAge() const
 {
-    return maxGameAgeComboBox->currentIndex();
+    int index = maxGameAgeComboBox->currentIndex();
+    if (index < 0 || index >= gameAgeMap.size()) { // index is out of bounds
+        return gamesProxyModel->getMaxGameAge();   // leave the setting unchanged
```

Comment:
if an alternative method for setting this value will arise (or the file is edited) this value won't be in the list so it's set to -1, this shows as an empty entry, while I don't think this will come up soon I think it's good to not have it crash then.